src/server/managers/AppApi.ts

```diff
@@ -0,0 +1,126 @@
+import { AppMethod } from '../../definition/metadata';
+
+import { ApiSecurity, ApiVisibility, IApi, IApiRequest, IApiResponse } from '../../definition/api';
+import { IApiEndpoint } from '../../definition/api/IApiEndpoint';
+import { IApiEndpointInfo } from '../../definition/api/IApiEndpointInfo';
+import { ProxiedApp } from '../ProxiedApp';
+import { AppLogStorage } from '../storage';
+import { AppAccessorManager } from './AppAccessorManager';
+
+const methods: Array<string> = [
+    'get',
+    'post',
+    'put',
+    'delete',
+    'head',
+    'options',
+    'patch',
+];
+
+export class AppApi {
+    public readonly computedPath: string;
+    public readonly basePath: string;
+    public readonly appId: string;
+    public readonly hash?: string;
+    public readonly implementedMethods: Array<string>;
+
+    constructor(public app: ProxiedApp, public api: IApi, public endpoint: IApiEndpoint) {
+        this.appId = app.getID();
+
+        switch (this.api.visibility) {
+            case ApiVisibility.PUBLIC:
+                this.basePath = `/api/apps/public/${app.getID()}`;
```

Comment:
@rodrigok what would you think about us making the `nameSlug` of the Apps to be unique and use it instead of the App's ID? This would help with the transition over to #56 (package.json) to slowly get rid of the usage of the id. As a result, and where the idea came to me from, the "endpoints" and urls generated/provided will be easier to use (and shorter) instead of having a long id in the url.

Reply:
I wan't to move the ID to be a unique ID based on name as NPM does, that way we keep using the ID here, just change how the ID would be generated.

Reply:
Right, so if we go with the name slug or a special smaller ID such as `author:app-name` then we could generate shorter urls and that would be a unique ID since the author couldn't have two of the same app names but two people could have the same App name as their author name would be different.

Reply:
Yes, but that will be in the future